docs/ci-cd-environment/setup-caching-on-aws-s3.md

```diff
@@ -0,0 +1,164 @@
+
+```
+aws s3api create-bucket \
+  --region us-east-1 \
+  --bucket semaphore-cache
+
+aws s3api put-public-access-block \
+  --bucket semaphore-cache \
+  --public-access-block-configuration \
+BlockPublicAcls=true,IgnorePublicAcls=true,BlockPublicPolicy=true,RestrictPublicBuckets=true
+```
+
+<b>2. Create an AWS IAM policy</b>
+
+It is always good practice to give services only the permissions they need. The cache CLI only needs access to perform the following AWS S3 actions:
+
+- `s3:PutObject`
+- `s3:GetObject`
+- `s3:ListBucket`
+- `s3:DeleteObject`
+
+One way to grant access to only these actions is through an AWS IAM policy. You can create one with the following commands:
+
+```
+cat > /tmp/semaphore-cache-policy.json <<EOF
+{
+  "Version": "2012-10-17",
+  "Statement": [
+    {
+      "Sid": "statement1",
+      "Effect": "Allow",
+      "Action": [
+        "s3:PutObject",
+        "s3:GetObject",
+        "s3:ListBucket",
+        "s3:DeleteObject"
+      ],
+      "Resource": [
+        "arn:aws:s3:::semaphore-cache/*",
+        "arn:aws:s3:::semaphore-cache"
+      ]
+    }
+  ]
+}
+EOF
+
+aws iam create-policy \
+  --policy-name semaphore-cache-policy \
+  --policy-document file:///tmp/semaphore-cache-policy.json \
+  --description "Restricts access to some operations on the semaphore-cache S3 bucket"
+```
+
+The `aws iam create-policy` command will generate an output like this:
+
+```json
+{
+  "Policy": {
+    "PolicyName": "semaphore-cache-policy",
+    "CreateDate": "2015-06-01T19:31:18.620Z",
+    "AttachmentCount": 0,
+    "IsAttachable": true,
+    "PolicyId": "ZXR6A36LTYANPAI7NJ5UV",
+    "DefaultVersionId": "v1",
+    "Path": "/",
+    "Arn": "arn:aws:iam::0123456789012:policy/semaphore-cache-policy",
+    "UpdateDate": "2015-06-01T19:31:18.620Z"
+  }
+}
+```
+
+Make sure you save the `Arn` of the AWS IAM policy. You'll use that value when attaching the AWS IAM policy to the AWS IAM user.
+
+<b>3. Create the AWS IAM user and its access keys:</b>
+
+This is the AWS IAM user the cache CLI will use to authorize and get access to your AWS bucket:
+
+```
+aws iam create-user --user-name semaphore
+aws iam create-access-key --user-name semaphore
+```
+
+The `aws iam create-access-key` command will generate an output like this:
+
+```json
+{
+  "AccessKey": {
+    "UserName": "semaphore",
+    "Status": "Active",
+    "CreateDate": "...",
+    "SecretAccessKey": "...",
+    "AccessKeyId": "..."
+  }
+}
+```
+
+Make sure you save the `SecretAccessKey` and the `AccessKeyId` values. We will need them when configuring the `~/.aws` folder in the agent machine.
+
+<b>4. Attach the AWS IAM policy to the AWS IAM user</b>
+
+Let's attach the AWS IAM policy we created on step #2 to the user we created on step #3:
+
+```
+aws iam attach-user-policy \
+  --user-name semaphore \
+  --policy-arn "[this is where the policy ARN you saved on step #2 needs to go]"
+```
+
+## Configure the agent machine
```

Comment:
just pls make sure that we use the same term here which also appears in the UI (in case it's maybe not "agent machine")